transport-native-unix-common/src/main/c/netty_unix_filedescriptor.c

```diff
@@ -163,13 +163,13 @@ static jlong netty_unix_filedescriptor_writev(JNIEnv* env, jclass clazz, jint fd
             (*env)->DeleteLocalRef(env, bufObj);
         }
     } else if (limitFieldId != NULL) {
-        for (i = offset; i < num; i++) {
+        for (i = offset; i < num; ++i) {
             jobject bufObj = (*env)->GetObjectArrayElement(env, buffers, i);
             jint pos = (*env)->CallIntMethod(env, bufObj, posId, NULL);
             jint limit = (*env)->GetIntField(env, bufObj, limitFieldId);
```

Comment:
What if you have a pending exception? Will this code work without calling `ExceptionOccurred/ExceptionClear`?

Reply:
We typically don't call `ExceptionOccurred` in JNI unless there is a specific need to. We generally let the exceptions be thrown (if any occurred) after transitioning back from JNI to Java.

Do you see a problem with this approach on this line/method, or just a general question?

Reply:
The exception will be thrown when you return from the method (if I remember correctly). I think JNI will keep returning 0s (or some other garbage) if you don't clear the exception. Think about what can happen when pos and limit are 0s (or a bad bufObj).

Reply:
Are you accounting for the case when JNI will start returning bad values because of an exception?

Reply:
It is more like a general question since it applies to all the JNI calls in netty, not that particular one.

Reply:
Lets consider this as a separate issue as the behavior w.r.t to JNI exception hasn't changed in this PR and scope is larger.  It may boil down to how paranoid/defensive do we want to be. Perhaps we should check the state before we make the system call.

Reply:
Thanks for bringing this up @todor-m ... see https://github.com/netty/netty/issues/7508